Neos.Flow/Classes/Persistence/Doctrine/Mapping/Driver/FlowAnnotationDriver.php

```diff
@@ -249,14 +249,7 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
             if ($tableAnnotation->uniqueConstraints !== null) {
                 foreach ($tableAnnotation->uniqueConstraints as $uniqueConstraint) {
                     $uniqueConstraint = ['columns' => $uniqueConstraint->columns];
-                    if (!empty($uniqueConstraint->options)) {
-                        $uniqueConstraint['options'] = $uniqueConstraint->options;
-                    }
-                    if (!empty($uniqueConstraint->name)) {
-                        $primaryTable['uniqueConstraints'][$uniqueConstraint->name] = $uniqueConstraint;
-                    } else {
-                        $primaryTable['uniqueConstraints'][] = $uniqueConstraint;
-                    }
+                    $primaryTable['uniqueConstraints'][] = $uniqueConstraint;
```

Comment:
Isnt that a change in bahavior as now `name` is not used as array index anymore? Also this would also transfer other keys like `fields` from the constraint. Might well be valid but i would like to unterstand the motivation.

Reply:
The problem was dead code. Phpstan (rightfully) detected that we foreach over `$uniqueConstraint` but in 251 we reassign the variable and thus the check `if (!empty($uniqueConstraint->options)) {` will always fail

the code was introduced as fix via https://github.com/neos/flow-development-collection/commit/012f0e76a3169ed9b30b2dd23698718c4d550782 but it never worked. So removing its seems the right choice? - I have no clue btw about this file ^^

Reply:
Extracted this less simple part into an own pr https://github.com/neos/flow-development-collection/pull/3263